### PR TITLE
[CL: Incentives]   Implement tick-level set and get functions for uptime trackers

### PR DESCRIPTION
**Pull request by stackman27**
[Closes: #4275](https://github.com/osmosis-labs/osmosis/issues/4275)

## What is the purpose of the change

Each tick currently needs to track something analogous to feeGrowthOutside for uptimes. Similar to incentive records, we need to store these in state in a way that is easily accessible (e.g. getAllUptimeTrackersForTick) while also being easy to abstract into a different module in the future.

## Brief Changelog
n/a

## Testing and Verifying
n/a

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/types/keys.go [hunk 1]
```diff
@@ -96,3 +96,7 @@ func KeyIncentiveRecord(poolId uint64, denom string, minUptime time.Duration) []
 func KeyPoolIncentiveRecords(poolId uint64) []byte {
 	return []byte(fmt.Sprintf("%s%s%d", IncentivePrefix, KeySeparator, poolId))
 }
+
+func KeyUptimeTracker(poolId uint64, tickIndex int64) []byte {
+	return []byte(fmt.Sprintf("%s%s%d%d", IncentivePrefix, KeySeparator, poolId, tickIndex))
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/types/keys.go:101]: Let's add another key separator here between `poolId` and `tickIndex`

File: x/concentrated-liquidity/incentives.go [hunk 2]
```diff
@@ -253,6 +254,32 @@ func (k Keeper) GetIncentiveRecord(ctx sdk.Context, poolId uint64, denom string,
...
+// nolint: unused
+// setUpTimeTracker sets the uptime tracker in state
+func (k Keeper) setUpTimeTracker(ctx sdk.Context, poolId uint64, tickIndex int64, uptimeTracker model.UptimeTracker) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.KeyUptimeTracker(poolId, tickIndex)
+	osmoutils.MustSet(store, key, &uptimeTracker)
+}
+
+// GetUpTimeTracker gets the uptime tracker corresponding to the passed in values from store
+func (k Keeper) GetUpTimeTracker(ctx sdk.Context, poolId uint64, tickIndex int64) (model.UptimeTracker, error) {
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/incentives.go:266]: nit: uncapitalize the T for naming consistency (for both `GetUpTimeTracker` and `setUpTimeTracker`)

**Review by AlpinYukseloglu — COMMENTED** [review]
The changes you made LGTM but we also need to include a "get all" function similar to this: https://github.com/osmosis-labs/osmosis/blob/dd76980944870d4fe165d5f91fc0a7da42475b00/x/concentrated-liquidity/incentives.go#L270

Feel free to reference the incentive record key/value parse function for how to do this efficiently

**Comment by stackman27** [comment]
Converting this to draft until we merge (https://github.com/osmosis-labs/osmosis/issues/4279) and https://github.com/osmosis-labs/osmosis/issues/4277

**Pull request closed without merging**